mailet/standard/src/test/java/org/apache/james/transport/mailets/RemoveMailAttributeTest.java

```diff
@@ -19,69 +19,101 @@
-                FakeMailContext.defaultContext());
-        if (attribute != null) {
-            mci.setProperty("name", attribute);
-        }
+    @Test
+    public void getMailetInfoShouldReturnCorrectInformation() throws Exception {
+        assertThat(removeMailet.getMailetInfo()).isEqualTo("Remove Mail Attribute Mailet");
+    }
 
-        mailet.init(mci);
-        return mailet;
+    @Test(expected = MailetException.class)
+    public void initShouldThrowExceptionIfMailetConfigDoesNotContainAttribute() throws MessagingException {
+        removeMailet.init(mailetConfig);
     }
 
+    @Test(expected = NullPointerException.class)
+    public void serviceShouldThrowExceptionWithMailNull() throws MessagingException {
+        removeMailet.service(null);
+    }
 
     @Test
-    public void testRemoveMailAttribute() throws MessagingException {
-        Mail m = createMail();
-        Mailet mailet = setupMailet(MAIL_ATTRIBUTE_NAME1);
-
-        // check if the mail has a attribute
-        Assert.assertNotNull("Attribute exists", m.getAttribute(MAIL_ATTRIBUTE_NAME1));
-        Assert.assertNotNull("Attribute exists", m.getAttribute(MAIL_ATTRIBUTE_NAME2));
+    public void serviceShouldDoNothingIfMailHasEmptyAttribute() throws MessagingException {
+        mailetConfig.setProperty(RemoveMailAttribute.MAILET_NAME_PARAMETER, ATTRIBUTE1_ATTRIBUTE2);
+        removeMailet.init(mailetConfig);
 
-        mailet.service(m);
+        mockMail = FakeMail.builder().build();
+        removeMailet.service(mockMail);
 
-        // Check if the attribute was removed
-        Assert.assertNull("Attribute exists", m.getAttribute(MAIL_ATTRIBUTE_NAME1));
-        Assert.assertNotNull("Attribute deleted", m.getAttribute(MAIL_ATTRIBUTE_NAME2));
+        assertThat(mockMail.getAttributeNames()).isEmpty();
     }
 
+    @Test
+    public void serviceShouldDoNothingIfMailDoNotMatchAttribute() throws MessagingException {
+        mailetConfig.setProperty(RemoveMailAttribute.MAILET_NAME_PARAMETER, ATTRIBUTE1_ATTRIBUTE2);
+        removeMailet.init(mailetConfig);
+
+        mockMail = FakeMail.builder().attribute(ATTRIBUTE_3, VALUE_3).build();
+        removeMailet.service(mockMail);
+
+        assertThat(mockMail.getAttributeNames()).containsExactly(ATTRIBUTE_3);
+    }
 
     @Test
-    public void testInvalidConfig() throws MessagingException {
-        boolean exception = false;
-        try {
-            setupMailet(null);
-        } catch (MessagingException e) {
-            exception = true;
-        }
-
-        Assert.assertTrue("invalid Config", exception);
+    public void serviceShouldRemoveSpecifiedAttribute() throws MessagingException {
+        mailetConfig.setProperty(RemoveMailAttribute.MAILET_NAME_PARAMETER, ATTRIBUTE_1);
+        removeMailet.init(mailetConfig);
+
+        mockMail = FakeMail.builder()
+            .attributes(ImmutableMap.<String, Serializable>of(ATTRIBUTE_1, VALUE_1, ATTRIBUTE_2, VALUE_2, ATTRIBUTE_3, VALUE_3))
```

Comment:
you should create a `attribute` method in FakeMail to define a single attribute at a time, removing the need to define a complex datastructure  upfront